devdoc/constbuffer_array_requirements.md

```diff
@@ -98,6 +98,34 @@ Note: `constbuffer_array_create_from_buffer_index_and_count` does not increment
 
 **SRS_CONSTBUFFER_ARRAY_42_016: [** If any error occurs then `constbuffer_array_create_from_buffer_index_and_count` shall fail and return `NULL`. **]**
 
+### constbuffer_array_create_from_start_and_end
+
+```c
+MOCKABLE_FUNCTION(, CONSTBUFFER_ARRAY_HANDLE, constbuffer_array_create_from_start_and_end, CONSTBUFFER_ARRAY_HANDLE, original, uint32_t, start_buffer_index, uint32_t, start_buffer_offset, uint32_t, start_buffer_size, uint32_t, end_buffer_index, uint32_t, end_buffer_offset, uint32_t, end_buffer_size);
+```
+
+`constbuffer_array_create_from_start_and_end` creates a new const buffer array which is a subset of the existing array in `original`.
+
+**SRS_CONSTBUFFER_ARRAY_07_001: [** If `original` is `NULL` then `constbuffer_array_create_from_start_and_end` shall fail and return `NULL`.  **]**
+
+**SRS_CONSTBUFFER_ARRAY_07_002: [** If `start_buffer_index` is greater than the number of buffers in `original` then `constbuffer_array_create_from_start_and_end` shall fail and return `NULL`.  **]**
+
+**SRS_CONSTBUFFER_ARRAY_07_003: [** If `end_buffer_index` is greater than the number of buffers in `original` then `constbuffer_array_create_from_start_and_end` shall fail and return `NULL`.  **]**
+
+**SRS_CONSTBUFFER_ARRAY_07_004: [** If `start_buffer_index` is greater than or equal to `end_buffer_index` then `constbuffer_array_create_from_start_and_end` shall fail and return `NULL`. **]**
```

Comment:
>or equal to  [](http://example.com/codeflow?start=76&length=12)

we should still allow this if the start buffer index == end buffer index (or count is 1).

Note that that means there is a special case for count==1 where the first and last buffer is really the same (and we need to handle that).

Reply:
done